src/containers/RemoveAppointments/RemoveAppointments.tsx

```diff
@@ -0,0 +1,182 @@
+  const params = queryString.parse(search);
+
+  const key = Array.isArray(params.externalKey)
+    ? params.externalKey[0]
+    : params.externalKey;
+
+  const {
+    data: locationData,
+    error: locationError,
+    loading: locationLoading,
+    refetch: locationRefetch,
+  } = useRetrieveLocationByExternalKeyApiV1LocationsExternalExternalKeyGet({
+    external_key: key || "",
+    lazy: true,
+  });
+
+  const {
+    data: availabilitiesData,
+    error: availabilitiesError,
+    loading: availabilitiesLoading,
+    refetch: availabilitiesRefetch,
+  } = useListVaccineAvailabilityLocationApiV1VaccineAvailabilityLocationGet({
+    queryParams: {
+      locationID: locationData?.id || -1,
+    },
+    lazy: true,
+  });
+
+  const {
+    mutate: put,
+    loading: updateAvailabilityLoading,
+    error: updateAvailabilityError,
+  } = useUpdateVaccineAvailabilityApiV1VaccineAvailabilityVaccineAvailabilityIdPut(
+    { vaccine_availability_id: avaibilityIdToUpdate },
+  );
+
+  // This will trigger on page load and will request location data
+  if (!locationLoading && !locationData && !locationError) {
+    locationRefetch().catch((err) => console.error(err));
+  }
+
+  // This will trigger once location data has ben loaded, and will fetch the
+  // availabilities in this location
+  const previousLocationData = usePrevious(locationData);
+  useEffect(() => {
+    if (!previousLocationData && locationData) {
+      availabilitiesRefetch().catch((err) => console.error(err));
+    }
+  }, [availabilitiesRefetch, locationData, previousLocationData]);
+
+  // Check to see if we have transitioned from loading to not loading of an availability update request
+  // - with the absence of an error - that indicates a PUT request succeeded and we should queue up our next
+  // availability to be processed if one exists
+  const previousUpdateAvailabilitiesLoading = usePrevious(
+    updateAvailabilityLoading,
+  );
+  useEffect(() => {
+    if (
+      previousUpdateAvailabilitiesLoading &&
+      !updateAvailabilityLoading &&
+      !updateAvailabilityError
+    ) {
+      if (availabilityIdsToUpdate.length > 0) {
+        const [first, ...rest] = availabilityIdsToUpdate;
+        setAvailabilityIdToUpdate(first);
+        setAvailabilityIdsToUpdate(rest);
+      } else {
+        setIsUpdateSuccessful(true);
+      }
+    }
+  }, [
+    availabilityIdsToUpdate,
+    previousUpdateAvailabilitiesLoading,
+    updateAvailabilityError,
+    updateAvailabilityLoading,
+  ]);
+
+  // If our current availability id transitions, a request has just been completed and we can
+  // update this new availability
+  const previousAvailabiltyIdToUpdate = usePrevious(avaibilityIdToUpdate);
+  useEffect(() => {
+    if (
+      previousAvailabiltyIdToUpdate !== avaibilityIdToUpdate &&
+      availabilitiesData
+    ) {
+      const availaibilityToUpdate = availabilitiesData.find(
+        (a) => a.id === avaibilityIdToUpdate,
+      )!;
+
+      put({
```

Comment:
I'm not entirely happy with this little dance where I maintain a queue of availability ids to process, and then update them one by one. Unfortunately I don't believe there is a way to make an unknown number of requests in parallel to the same endpoint as `restful-react` binds parameters into the hook on each render, and we cannot create multiple mutators as hooks cannot be placed into loops.

Theoretically I suppose I could create a component that wraps the hook, and then add a component array to this component. Don't know if that would simplify things.